### PR TITLE
feat: revise themes, publishers and about pages

### PR DESCRIPTION
**Pull request by zalborzi**
Screenshot 2024-08-16 at 14 27 04
Screenshot 2024-08-16 at 14 27 12
Screenshot 2024-08-16 at 14 27 20

## Summary by Sourcery

Revise the About, Themes, and Organizations pages to improve content loading and user interface. Implement dynamic markdown content loading for the About page, update the OrganizationList component layout, and enhance the Themes and Organizations pages with item count badges.

New Features:
- Introduce dynamic content loading for the About page using markdown fetched from a server.

Enhancements:
- Redesign the About page to use a new renderer for markdown content, improving the flexibility and maintainability of the page.
- Update the OrganizationList component to improve layout and styling, including changes to grid columns and item dimensions.
- Revise the Themes and Organizations pages to include a badge displaying the count of items, enhancing the user interface.

### CODE REVIEW

File: src/app/about/renderer.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/* SPDX-FileCopyrightText: 2024 PNED G.I.E. */
+
+/* SPDX-License-Identifier: Apache-2.0 */
+
+import { Renderer, marked } from "marked";
```
Comment (brunopacheco1) [inline src/app/about/renderer.ts:5]: question: is this file still needed? It looks like it's not used anywhere.

File: public/about.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+vast repository of over one million genome sequences, this platform is
+currently under development and will serve as the main European-level
+hub for data access, providing a user-friendly interface for
+researchers and healthcare professionals.
+
+## Key Objectives
+
+- To link and to provide cross-border access to genomic and related phenotypic datasets across Europe
+- To advance understanding of genomics for more precise and faster clinical decision-making, diagnostics, treatments systems, and to benefit the overall economy
+- To align with the development under the European Health Data Space (EHDS)
```
[the remark this answers is not in the document]
Comment (zalborzi) [reply public/about.md:25]: I am not going to change its content.

File: public/about.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+as well as access to a reference genome collection representing the
+European population (Genome of Europe).
+
+## User Portal
+
+The User Portal, developed by the Genomic Data Infrastructure (GDI)
+project, is the central entry point for accessing genomic data. As
+part of the Genomic Data Infrastructure (GDI) project, it unlocks a
+vast repository of over one million genome sequences, this platform is
+currently under development and will serve as the main European-level
```
[the remark this answers is not in the document]
Comment (zalborzi) [reply public/about.md:17]: I am not going to change its content.

File: src/components/OrganizationList.tsx [hunk 4]
```diff
@@ -15,15 +15,18 @@ const OrganizationList: React.FC<OrganizationListProps> = ({
...
+      {/* Added bottom margin */}
       <div className="flex flex-col w-full">
-        <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-8 justify-center">
+        <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-5 gap-4 justify-center">
           {organizations.map((org) => (
             <div
               key={org.id}
-              className="bg-white py-6 flex flex-col items-start justify-start rounded-lg shadow-lg border-b-4 border-b-[#B5BFC4] hover:border-b-secondary transition hover:bg-gray-50 text-left"
+              className="bg-white py-4 flex flex-col items-start justify-start rounded-lg shadow-lg border-b-4 border-b-[#B5BFC4] hover:border-b-secondary transition hover:bg-gray-50 text-left"
+              style={{ maxWidth: "260px" }}
```
[the remark this answers is not in the document]
Comment (zalborzi) [reply src/components/OrganizationList.tsx:27]: it was there, it stays there.

File: src/app/about/page.tsx [hunk 5]
```diff
@@ -4,89 +1,77 @@
...
+  }
+
+  return (
+    <PageContainer className="container mx-auto px-4 pt-5">
+      <div className="my-8 flex items-center gap-2">
+        <h1 className="text-left font-medium text-2xl sm:text-3xl">About</h1>
+      </div>
+      <div
+        className="text-base leading-relaxed"
+        dangerouslySetInnerHTML={{ __html: response.content || "" }}
```
[the remark this answers is not in the document]
Comment (zalborzi) [reply src/app/about/page.tsx:72]: Good! should I use DOMPurify? would that solve the issue?

File: src/app/about/page.tsx [hunk 6]
```diff
@@ -1,92 +1,76 @@
-// SPDX-FileCopyrightText: 2024 PNED G.I.E.
-//
-// SPDX-License-Identifier: Apache-2.0
+/* SPDX-FileCopyrightText: 2024 PNED G.I.E. */
 
+/* SPDX-License-Identifier: Apache-2.0 */
+
+"use client";
```
Comment (brunopacheco1) [inline src/app/about/page.tsx:5]: question: @zalborzi why does it have to be a client component? You can access the about.md directly in the server side.

File: src/app/about/page.tsx [hunk 7]
```diff
@@ -1,92 +1,76 @@
...
-          <li>
-            To maintain awareness, acceptance and trust in the main groups of
-            stakeholders, notably European citizens, data holders, healthcare
-            professionals, researchers and public health authorities
-          </li>
-        </ul>
-      </p>
+        // @ts-expect-error" Type assertion is necessary because our custom renderer is a partial
+        // implementation of the Renderer interface, but we know it's compatible for our use case.
+        marked.use({ renderer: renderer as Partial<marked.Renderer> });
```
Comment (brunopacheco1) [inline src/app/about/page.tsx:36]: question: why do we need to have a special renderer?

File: src/app/about/page.tsx [hunk 8]
```diff
@@ -1,92 +1,77 @@
...
-          project, is the central entry point for accessing genomic data. As
-          part of the Genomic Data Infrastructure (GDI) project, it unlocks a
-          vast repository of over one million genome sequences, this platform is
-          currently under development and will serve as the main European-level
-          hub for data access, providing a user-friendly interface for
-          researchers and healthcare professionals.{" "}
-        </p>
-      </p>
+  useEffect(() => {
+    const fetchMarkdown = async () => {
```
[the remark this answers is not in the document]
Comment (brunopacheco1) [reply src/app/about/page.tsx:28]: @zalborzi it is a reasonable comment, can you please check it? It's related to fetch the markdown file.

**Review by brunopacheco1 — DISMISSED** [review]

File: src/app/about/page.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (brunopacheco1) [reply src/app/about/page.tsx:72]: important: @zalborzi  please check this comment.

**Comment by brunopacheco1** [comment]
important: @zalborzi please run `npm audit fix`, to update `package-lock.json`.

image

File: src/app/about/page.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (zalborzi) [reply src/app/about/page.tsx]: isomorphic-dompurify as i changed it to be in server side.

File: src/app/about/page.tsx (hunk 6 above)
Comment (zalborzi) [reply src/app/about/page.tsx]: I went with my gut! I faced several issues. so i changed it to be server side.

File: src/app/about/page.tsx (hunk 7 above)
Comment (zalborzi) [reply src/app/about/page.tsx]: to show the tags content based on what we want to show... with styles and ...

File: src/app/about/page.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (zalborzi) [reply src/app/about/page.tsx]: it is not the case any more as as it is in server side rendering now.

**Review by nolliia — APPROVED** [review]

**Pull request merged**